opsmngr/backup_administrator.go

```diff
@@ -0,0 +1,874 @@
+	URI                  string   `json:"uri,omitempty"`
+	WriteConcern         string   `json:"writeConcern,omitempty"`
+	Labels               []string `json:"labels,omitempty"`
+	SSL                  bool     `json:"ssl"`
+	AssignmentEnabled    bool     `json:"assignmentEnabled"`
+	EncryptedCredentials bool     `json:"encryptedCredentials"`
+	UsedSize             int64    `json:"usedSize,omitempty"`
+}
+
+// S3Blockstore represents a S3Blockstore in the MongoDB Ops Manager API
+type S3Blockstore struct {
+	Blockstore
+	AWSAccessKey           string `json:"awsAccessKey,omitempty"`
+	AWSSecretKey           string `json:"awsSecretKey,omitempty"`
+	DisableProxyS3         string `json:"disableProxyS3,omitempty"`
+	S3AuthMethod           string `json:"s3AuthMethod,omitempty"`
+	S3BucketEndpoint       string `json:"s3BucketEndpoint,omitempty"`
+	S3BucketName           string `json:"s3BucketName,omitempty"`
+	S3MaxConnections       int64  `json:"s3MaxConnections,omitempty"`
+	AcceptedTos            bool   `json:"acceptedTos"`
+	SSEEnabled             bool   `json:"sseEnabled"`
+	PathStyleAccessEnabled bool   `json:"pathStyleAccessEnabled"`
+}
+
+// S3Blockstores represents a paginated collection of S3Blockstore
+type S3Blockstores struct {
+	Links      []*atlas.Link   `json:"links"`
+	Results    []*S3Blockstore `json:"results"`
+	TotalCount int             `json:"totalCount"`
+}
+
+// Blockstore represents a Blockstore in the MongoDB Ops Manager API
+type Blockstore struct {
+	AdminConfig
+	LoadFactor    int64  `json:"loadFactor,omitempty"`
+	MaxCapacityGB int64  `json:"maxCapacityGB,omitempty"`
+	Provisioned   bool   `json:"provisioned"`
+	SyncSource    string `json:"syncSource,omitempty"`
+	Username      string `json:"username,omitempty"`
+}
+
+// Blockstores represents a paginated collection of Blockstore
+type Blockstores struct {
+	Links      []*atlas.Link `json:"links"`
+	Results    []*Blockstore `json:"results"`
+	TotalCount int           `json:"totalCount"`
+}
+
+// FileSystemStoreConfiguration represents a File System Store Configuration in the MongoDB Ops Manager API
+type FileSystemStoreConfiguration struct {
+	AdminConfig
+	LoadFactor               int64  `json:"loadFactor,omitempty"`
+	MMAPV1CompressionSetting string `json:"mmapv1CompressionSetting,omitempty"`
+	StorePath                string `json:"storePath,omitempty"`
+	WTCompressionSetting     string `json:"wtCompressionSetting,omitempty"`
+	AssignmentEnabled        bool   `json:"assignmentEnabled"`
+}
+
+// FileSystemStoreConfigurations represents a paginated collection of FileSystemStoreConfiguration
+type FileSystemStoreConfigurations struct {
+	Links      []*atlas.Link                   `json:"links"`
+	Results    []*FileSystemStoreConfiguration `json:"results"`
+	TotalCount int                             `json:"totalCount"`
+}
+
+// Oplog represents a Oplog Configuration in the MongoDB Ops Manager API
+type Oplog struct {
+	Blockstore
+}
+
```

Comment:
[q] do we need a new type that's the same as `Blockstore`? why not just use `Blockstore`?

Reply:
I thought to create a new struct for the following reasons 

1. I thought that using `Blockstore` with the functions inside `OplogService` may be confusing
2. The `oplog` may have new fields in the future that are not inside `Blockstore`

Reply:
for 1, then we can pick a more generic name, `BackupStore` or something like that, for 2, that's just premature optimisation